build/helper.py

```diff
@@ -240,3 +241,121 @@ def get_rst_header_snippet(t, header_level='='):
+
+    >>> print as_rest_table(data)
+    =====  ====================  ======
+    what   how                   who   
+    =====  ====================  ======
+    lorem  that is a long value  3.1415
+    ipsum                 89798     0.2
+    =====  ====================  ======
+
+    """
+    data = data if data else [['No Data']]
+    table = []
+    # max size of each column
+    sizes = map(max, zip(*[[len(str(elt)) for elt in member]
+                           for member in data]))
+    if sys.version_info.major >= 3:
+        sizes = list(sizes)
+    num_elts = len(sizes)
+
+    if full:
+        start_of_line = '| '
+        vertical_separator = ' | '
+        end_of_line = ' |'
+        line_marker = '-'
+    else:
+        start_of_line = ''
+        vertical_separator = '  '
+        end_of_line = ''
+        line_marker = '='
+
+    meta_template = vertical_separator.join(['{{{{{0}:{{{0}}}}}}}'.format(i)
+                                             for i in range(num_elts)])
+    template = '{0}{1}{2}'.format(start_of_line,
+                                  meta_template.format(*sizes),
+                                  end_of_line)
+    # determine top/bottom borders
+    if full:
+        to_separator = {ord('|'): '+', ord(' '): '-'}
+        if sys.version_info.major < 3:
+            to_separator = string.maketrans('| ', '+-')
+    else:
+        to_separator = {ord('|'): '+'}
+        if sys.version_info.major < 3:
+            to_separator = string.maketrans('|', '+')
+    start_of_line = start_of_line.translate(to_separator)
+    vertical_separator = vertical_separator.translate(to_separator)
+    end_of_line = end_of_line.translate(to_separator)
+    separator = '{0}{1}{2}'.format(start_of_line,
+                                   vertical_separator.join([x*line_marker for x in sizes]),
+                                   end_of_line)
+    # determine header separator
+    th_separator_tr = {ord('-'): '='}
+    if sys.version_info.major < 3:
+        th_separator_tr = string.maketrans('-', '=')
+    start_of_line = start_of_line.translate(th_separator_tr)
+    line_marker = line_marker.translate(th_separator_tr)
+    vertical_separator = vertical_separator.translate(th_separator_tr)
+    end_of_line = end_of_line.translate(th_separator_tr)
+    th_separator = '{0}{1}{2}'.format(start_of_line,
+                                      vertical_separator.join([x*line_marker for x in sizes]),
+                                      end_of_line)
+    # prepare result
+    table.append(separator)
+    # set table header
+    titles = data[0]
+    table.append(template.format(*titles))
+    table.append(th_separator)
+
+    for d in data[1:-1]:
+        table.append(template.format(*d))
+        if full:
+            table.append(separator)
+    table.append(template.format(*data[-1]))
+    table.append(separator)
+    return '\n'.join(table)
+
+def get_python_type_from_visa_type(visa_type):
```

Comment:
I thought about that and immediately saw two problems. 
1. This is a different context than the classes we have. Those are for during runtime of the driver, this is for build time when we are generating the driver. While I think it would work, it seems odd to include something from the templates folder during build. 
2. I am not instantiating any of these to then query the base type. I would need something like an orb (object request broker) to instantiate an object from just a name.